pkgs/development/compilers/rust/rls/default.nix

```diff
@@ -14,7 +14,7 @@ rustPlatform.buildRustPackage {
 
   preBuild = ''
     # client tests are flaky
-    rm tests/client.rs
+    rm $buildAndTestSubdir/tests/client.rs
```

Comment:
That is not very intuitiv.

Reply:
Could be `$sourceRoot` used here?

Reply:
I may miss the context so, why we have this variable in the first place.

Reply:
I think we can't because cargo needs the whole workspace to download dependencies.

Reply:
The variable is defined above, should this not clear up things?

Reply:
@SuperSandro2000 if you prefer, I can also make the attrset recursive and use `${buildAndTestSubdir}` would that make it clearer?

Reply:
Now uses string interpolation to make this a bit more explicit. Does that address your comment?